paf-mvp-audit/src/model.ts

```diff
@@ -1,10 +1,57 @@
-import { Field, IFieldBind, IModel } from '@core/ui/fields';
+import { Field, FieldReadOnly, IFieldBind, IModel } from '@core/ui/fields';
 import { AuditLog, TransmissionResult } from '@core/model/generated-model';
+import { GetIdentityResponse } from '@core/model/generated-model';
+import { FileExtensionInfo } from 'typescript';
 
 /**
- * Field represents the transmission result from the audit log.
+ * Represents a transmission result from the audit log with additional features to support a hierarchy, fetching
+ * identity data, and verifying the signature.
  */
-export class FieldTransmissionResult extends Field<TransmissionResult, Model> implements IFieldBind {}
+export class TransmissionResultNode {
+  /**
+   * Cached copy of the identity information for the domain associated with the result.
+   */
+  private identity: GetIdentityResponse = undefined;
+
+  /**
+   * The result returned in the audit log.
+   */
+  public readonly result: TransmissionResult;
+
+  /**
+   * Constructs a new instance of TransmissionResultNode for the audit log record provided.
+   */
+  constructor(result: TransmissionResult) {
+    this.result = result;
+  }
+
+  /**
+   * Identity of the transmission result.
+   * @remarks this method gets the identity and all the public keys in a single method to avoid fetching the public key
+   * and the other identity information over multiple requests to the same endpoint.
+   * @returns promise that when resolved will provide the identity associated with the transmission result or an error.
+   */
+  public getIdentity() {
```

Comment:
NIT: return type

Reply:
Would you mind keeping it separate instead?
This way it would be easier to review: in the current review I think there are only a couple of files that "need" to be reviewed by Romain, the rest is mostly about translation, identity and pattern library.
I'm wondering if adding "quite a few changes" to this review would not make it complicated to review.